tests/test_runner.py

```diff
@@ -69,3 +71,35 @@ def test_add_first_appeared_dates():
     }
 
     assert runner.add_first_appeared_dates(probes_by_channel, first_appeared_dates) == expected
+
+
+def test_trailing_space():
```

Comment:
This test dumps files in the current directory, which is generally not a good idea.  Let's use the [tmp_path](https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture) fixture here instead.

Reply:
Okay sir I will use tmp_path intead